lib/rules/forbid-component-props.js

```diff
@@ -95,9 +113,18 @@ module.exports = {
                     minItems: 1,
                     items: { type: 'string' },
                   },
+                  disallowedForPatterns: {
+                    type: 'array',
+                    uniqueItems: true,
+                    minItems: 1,
+                    items: { type: 'string' },
+                  },
                   message: { type: 'string' },
                 },
-                required: ['disallowedFor'],
+                anyOf: [
```

Comment:
```suggestion
                oneOf: [
```
this one too?

Reply:
the same https://github.com/jsx-eslint/eslint-plugin-react/pull/3805#discussion_r1773456872